extensions.md

```diff
@@ -186,3 +186,53 @@ the server, but LSP does not expose them.
   - Clients that don't support client-side filtering should ignore this and
     use `sortText`, which incorporates `score` and fuzzy-matching.
 
+## AST
+{:v12}
+
+C++ has a complicated grammar and semantic structure that's not always obvious
+from the source. Inspecting the clang AST can lend some insight.
+Despite the name, it captures semantics as well as syntax (e.g. implicit casts).
+
+**New client->server request**: `textDocument/ast`:
+ - Params: `ASTParams` object with properties:
+   - `textDocument : TextDocumentIdentifier`: the open file to inspect
+   - `range : Range`: identifies the AST node to fetch.
+     The highest-level node that entirely contains the range is returned.
```

Comment:
Maybe reorder these two sentences? The first one seems strange (I'm reading `Range` - the AST node to fetch, and clarification comes later).

Reply:
Rephrased the first sentence to make it more explicit instead.
(The first is intended to give the general idea: select some code, that's what the returned AST will describe. The second makes this more precise.)